File: apps/cli/internal/adapter/sqlite/project_types.go
package sqlite

// ProjectCommand is a named shell command configured for a project.
type ProjectCommand struct {
	Name    string `json:"name"`
	Command string `json:"command"`
}

// Project is the daemon-owned local project record.
type Project struct {
	ID              string           `json:"id"`
	Name            string           `json:"name"`
	TaskPrefix      *string          `json:"taskPrefix"`
	SourceType      string           `json:"sourceType"`
	RepoProvider    *string          `json:"repoProvider,omitempty"`
	RepoURL         *string          `json:"repoUrl,omitempty"`
	RepoKey         *string          `json:"repoKey,omitempty"`
	Icon            string           `json:"icon"`
	Color           string           `json:"color"`
	SetupScript     string           `json:"setupScript"`
	PostScript      string           `json:"postScript"`
	Commands        []ProjectCommand `json:"commands"`
	ContextEnabled  bool             `json:"contextEnabled"`
	OrganizationID  string           `json:"organizationId"`
	CreatedByUserID *string          `json:"createdByUserId,omitempty"`
	CreatedAt       string           `json:"createdAt"`
	UpdatedAt       string           `json:"updatedAt"`
}

File: apps/desktop/src/renderer/domains/project/projectTypes.ts
/**
 * Project feature vocabulary (Phase 3 split of features/workbench/types.ts).
 */
export type WorkspaceProjectCommand = {
  name: string;
  command: string;
};

export type WorkspaceProjectRecord = {
  id: string;
  organizationId?: string;
  name: string;
  key?: string;
  path?: string;
  missing?: boolean;
  gitUrl?: string;
  sourceType?: "git" | "git-local" | "unknown";
  repoProvider?: string | null;
  repoUrl?: string | null;
  repoKey?: string | null;
  localPath?: string | null;
  worktreePath?: string | null;
  contextEnabled?: boolean;
  taskPrefix?: string | null;
  defaultBranch?: string | null;
  icon?: string | null;
  color?: string | null;
  setupScript?: string | null;
  postScript?: string | null;
  commands?: WorkspaceProjectCommand[];
  createdAt?: string;
  updatedAt?: string;
  createdByUserId?: string;
};

import type { ExternalAppId } from "../../../shared/contracts/externalApps";

/**
 * Synthetic project id used for local (non-git) folder workspaces. Folder
 * workspaces are daemon-owned rows (kind="folder") mapped into the workspace
 * list but have no real backend project, so they share this sentinel value.
 */
export type WorkspaceStoreOrganizationPreference = {
  displayProjectIds?: string[];
  knownProjectIds?: string[];
  lastUsedExternalAppId?: ExternalAppId;
};
